### PR TITLE
Start GoCD Server in maintenance mode. (#5787)

### PR DESCRIPTION
**Pull request by GaneshSPatil**
* Introduce 'gocd.server.start.in.maintenance.mode' system property
  to specify starting GoCD Server in maintenance mode.

* Update maintenance mode message when server is started in maintenance
  mode on:
	- maintenance mode spa.
	- global page footer.
	- maintenance mode banner at server_in_maintenance_mode.html

### CODE REVIEW

File: base/src/main/java/com/thoughtworks/go/util/SystemEnvironment.java [hunk 1]
```diff
@@ -799,6 +801,10 @@ public double getMDUExponentialBackOffMultiplier() {
         return MDU_EXPONENTIAL_BACKOFF_MULTIPLIER.getValue();
     }
 
+    public boolean shouldStartInMaintenanceMode() {
```
Comment (maheshp) [inline base/src/main/java/com/thoughtworks/go/util/SystemEnvironment.java:804]: `shouldStartInMaintenanceMode` => `shouldStartServerInMaintenanceMode`?

**Review by maheshp — APPROVED** [review]
LGTM, looks like tests are failing. Please take a look.

File: base/src/main/java/com/thoughtworks/go/util/SystemEnvironment.java (hunk 1 above)
Comment (GaneshSPatil) [reply base/src/main/java/com/thoughtworks/go/util/SystemEnvironment.java]: Done!

**Comment by maheshp** [comment]
Verified on `GoCD Version: 20.9.0 (12264-30bf28b07277798dfc36761dad864e170cee5336)`

**Pull request merged**